packages/streamr-logger/include/streamr-logger/Logger.hpp

```diff
@@ -1,18 +1,152 @@
+    char* val = getenv(envLogLevelName);
+    if (val) {
+        auto pair = toFollyLogLevelMap.find(val);
+        if (pair != toFollyLogLevelMap.end()) {
+            return std::optional<folly::LogLevel>{std::move(pair->second)};
+        }
+    }
+    return std::nullopt;
+}
+
+}; // namespace
 
-namespace streamr::logger {
 class Logger {
 public:
-    Logger() = default;
 
-    // NOLINTNEXTLINE(readability-convert-member-functions-to-static)
-    void log(const std::string& message) const {
-        std::cout << message << std::endl;
+    Logger(std::shared_ptr<folly::LogWriter> logWriter = nullptr)
+        : loggerDB_{folly::LoggerDB::get()} {
+        if (logWriter) {
+            writerFactory_ = StreamrWriterFactory(logWriter);
+        } else {
+            writerFactory_ = StreamrWriterFactory();
+        }
+        logHandlerFactory_ = {
+            std::make_unique<StreamrHandlerFactory>(&writerFactory_)};
+        this->initializeLoggerDB(folly::LogLevel::DBG);
+    }
+
+    static Logger& get(std::shared_ptr<folly::LogWriter> logWriter = nullptr) {
+        static Logger instance(logWriter);
+        return instance;
+    }
+
+    void log(StreamrLogLevel streamrLevel, const std::string& message) {
+        auto follyRootLogLevel = getFollyLogLevelFromEnv();
+        auto pair = fromStreamrToFollyLogLevelMap.find(streamrLevel);
+        if (pair == fromStreamrToFollyLogLevelMap.end()) {
+            // Not found from map, return
+            return;
+        }
+        if (follyRootLogLevel) {
+            if (follyRootLogLevel != loggerDB_.getCategory("")->getLevel()) {
+                // loggerDB.setLevel("", *follyLogLevel);
+                this->initializeLoggerDB(*follyRootLogLevel, true);
+            }
+            // This complicated code is mostly copy/pasted from folly's XLOG
+            // macro. It is not very flexible to call macro from here so. This
+            // code below prints a log message without any return code.
+            folly::LogStreamProcessor(
+                [] {
+                    static ::folly::XlogCategoryInfo<XLOG_IS_IN_HEADER_FILE>
+                        folly_detail_xlog_category;
+                    return folly_detail_xlog_category.getInfo(
+                        &::folly::detail::custom::xlogFileScopeInfo);
+                }(),
+                (pair->second),
+                [] {
+                    constexpr auto* folly_detail_xlog_filename = XLOG_FILENAME;
+                    return ::folly::detail::custom::getXlogCategoryName(
+                        folly_detail_xlog_filename, 0);
+                }(),
+                ::folly::detail::custom::isXlogCategoryOverridden(0),
+                XLOG_FILENAME,
+                __LINE__,
+                __func__,
+                ::folly::LogStreamProcessor::APPEND,
+                message)
+                .stream();
+        }
+    }
+
+private:
+    std::shared_ptr<folly::LogWriter> logWriter_;
```

Comment:
do not use underscore in member variable names

Reply:
What is the naming convention for member variables in this project. This is the way what is used in Folly. One way is to use hungarian notation. There are many different ways.